### ISSUE TITLE
RuntimeError when loading model checkpoint

### ISSUE THREAD

**Issue by sohamtiwari3120** [s3prl/s3prl#197]
Hi @andi611 @leo19941227 

When I am trying to load the model state dict for my pytorch model, I am getting the following error:
```
  File "/opt/conda/lib/python3.8/site-packages/torch/nn/modules/module.py", line 1406, in load_state_dict
    raise RuntimeError('Error(s) in loading state_dict for {}:\n\t{}'.format(
RuntimeError: Error(s) in loading state_dict for CustomModel:
        size mismatch for extractor_s3prl.extracter._melscale.fb: copying a param with shape torch.Size([513, 64]) from checkpoint, the shape in current model is torch.Size([0]).
```

Can you guide me on how to resolve the same?

Below is just a snippet roughly representing my model code for your reference. After training my model, I saved its model state.
```
class CustomModel(nn.Module):
    def __init__(self,  ...):

        super(CustomModel, self).__init__()

        self.s3prl_feature_type = 'baseline_local'
        self.extractor_s3prl =  torch.hub.load('s3prl/s3prl', self.s3prl_feature_type, model_config='path_to_custom_mel.yaml').to('cuda')
```

Sincerely,
Soham Tiwari

**Comment by leo19941227** [comment]
Hi @sohamtiwari3120,

Can you provide a minimal script for me to reproduce the issue?
Thanks!

**Comment by sohamtiwari3120** [comment]
Hi @leo19941227, I have provided the minimal script below, hope it is helpful

```
import torch
import torch.nn as nn

class CustomModel(nn.Module):
    def __init__(self):

        super(CustomModel, self).__init__()
            
        self.s3prl_feature_type = 'baseline_local'
        self.extractor_s3prl =  torch.hub.load('s3prl/s3prl', self.s3prl_feature_type, model_config='mel.yaml').to('cuda')

    def forward(self, x):

        wavs = [wav.to('cuda') for wav in x.squeeze(1)]
        out_s3prl = self.extractor_s3prl(wavs)['last_hidden_state']
        x = torch.unsqueeze(out_s3prl, 1)

        output_dict = {
            'x': x
        }

        return output_dict
    
model = CustomModel()
x = torch.rand([1, 192000]).to('cuda')
out = model(x)
checkpoint = {
    'model': model.state_dict()
}
torch.save(checkpoint, 'custom_model.pth')

# loading state dict
model = CustomModel()
checkpoint = torch.load('custom_model.pth')
model.load_state_dict(checkpoint['model'])
```

mel.yaml
```
win_ms: 64
hop_ms: 20
n_freq: 513 
n_mels: 64
n_mfcc: 13

input:
  feat_type: mel # feat_type can be: wav, complx, linear, mel, mfcc, phase
  channel: 0
  log: True
  delta: 0
  cmvn: True
```

**Comment by sohamtiwari3120** [comment]
Hi @leo19941227  @andi611
Do you have any workarounds I can try out for the same?

**Comment by andi611** [comment]
It looks like a torchaudio problem.
See the [MelScale](https://pytorch.org/audio/stable/_modules/torchaudio/transforms.html#MelScale) source code:
```python
fb = torch.empty(0) if n_stft is None else F.create_fb_matrix(
            n_stft, self.f_min, self.f_max, self.n_mels, self.sample_rate, self.norm, self.mel_scale)
self.register_buffer('fb', fb)
```
where `n_stft` is described as follow:
```
n_stft (int, optional): Number of bins in STFT. Calculated from first input
            if None is given.  See ``n_fft`` in :class:`Spectrogram`. (Default: ``None``)
```
Can you try forwarding the model with pseudo input, before you load the weight? The first input should register the buffer.

**Comment by andi611** [comment]
@leo19941227 I found this part in the source code:

```pyhton
if n_stft is None or n_stft == 0:
    warnings.warn(
        'Initialization of torchaudio.transforms.MelScale with an unset weight '
        '`n_stft=None` is deprecated and will be removed in release 0.10. '
        'Please set a proper `n_stft` value. Typically this is `n_fft // 2 + 1`. '
        'Refer to https://github.com/pytorch/audio/issues/1510 '
        'for more details.'
```
Should we pass an `n_stft` value when we initialize `MelScale`?

**Comment by leo19941227** [comment]
Cool, it seems to be the torchaudio version problem.
Yes I think we could upgrade the code to accommodate the latest usage.
Thanks for tracing the problem!

**Comment by leo19941227** [comment]
Hey actually this is torchaudio's bug and will be resolved after `torchaudio==v0.10.0`
You can refer to [this issue](https://github.com/pytorch/audio/issues/1510) and I provide a workaround below with adding just one line:

```
import torch
import torch.nn as nn
from s3prl.hub import baseline_local

class CustomModel(nn.Module):
    def __init__(self):

        super(CustomModel, self).__init__()

        self.extractor_s3prl = baseline_local(model_config="mel.yaml").to("cuda") 

    def forward(self, x):

        wavs = [wav.to('cuda') for wav in x.squeeze(1)]
        out_s3prl = self.extractor_s3prl(wavs)['last_hidden_state']
        x = torch.unsqueeze(out_s3prl, 1)

        output_dict = {
            'x': x
        }

        return output_dict
    
model = CustomModel()
x = torch.rand([1, 192000]).to('cuda')
out = model(x)
checkpoint = {
    'model': model.state_dict()
}
torch.save(checkpoint, 'custom_model.pth')

# loading state dict
model = CustomModel()

# ADD THIS LINE
out = model(x)

checkpoint = torch.load('custom_model.pth')
model.load_state_dict(checkpoint['model'])
```
